sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -469,6 +469,35 @@ def test_empty_output_box(gallery_conf):
     assert output.isspace()
 
 
+@pytest.mark.parametrize('code,out', [
+    ('a=2\nb=3', ''),
+    ('a=2\na', '2'),
+    ('a=2\nprint(a)', '2'),
+    ('print("hello")\na=2\na', 'hello\n\n2')])
```

Comment:
You could also parametrize over whether `print_eval_repr` is `('__repr__',)` or `()` and triage the `output_test_string` check below based on (in the empty case) whether the last line contains `print()` or not

Reply:
Sorry, I wasn't quite sure if you meant to parametrize twice, e.g.:

```Python
@pytest.mark.parametrize('config', [(),('__repr__',)])
@pytest.mark.parametrize('code', ['code1','code2'])
```
I tried to do this but couldn't figure out a good way because the output for all options (2x2 in the case above) was different - the only solution I thought of was to write out all combinations. Also I wasn't sure what you meant by triage...

(I will explain the tests I made below)

Reply:
You can do just pairs rather than 2x2 with:
```
pytest.mark.parametrize('config, code', [((), 'code1'), (('__repr__',), 'code2')])
```